server/src/main/webapp/WEB-INF/rails/webpack/models/materials/types.ts

```diff
@@ -51,6 +51,24 @@ export const mapTypeToDisplayType: { [key: string]: string; } = {
   plugin:     "Plugin"
 };
 
+function clean(url?: string) {
+  if (!url) {
+    return url;
+  }
+  const parsedUrl = urlParse(url, {});
+  if (!_.isEmpty(parsedUrl.auth)) {
+    if (parsedUrl.auth.includes(":")) {
+      parsedUrl.set('password', '******');
+      return parsedUrl.href;
+    }
+    if (parsedUrl.protocol.includes('ssh') || parsedUrl.protocol.includes('svn+ssh')) {
+      return parsedUrl.href;
+    }
+    parsedUrl.set('username', '******');
+  }
+  return parsedUrl.href;
+}
+
```

Comment:
* how about renaming the method name `urlForDisplay`?

* How about simplifying the above code as:
```javascript
function urlForDisplay(url?: string) {
  if (!url) { return undefined;}

  const parsed = urlParse(url, {});
  // do not mask passwords
  if (_.isEmpty(parsed.auth) || parsed.protocol.includes('ssh') || parsed.protocol.includes('svn+ssh')) {
    return parsed.href;
  }

  if (parsed.auth.includes(":")) {
    // mask password when username and password is provided as username:password@url
    parsed.set('password', '******');
  } else {
    // mask token when credentials are provided as token@url
    parsed.set('username', '******');
  }

  return parsed.href;
}
```